### ISSUE TITLE
Webhook error in 0.24

### ISSUE THREAD

**Issue by apranda** [JimStar/reolink_cctv#47]
After updating to 0.24, the sensors on my doorbell camera stopped updating in Home Assistant. I tried to fix this by removing the camera and then adding it back, however the initial configuration screen returned an "Unexpected error" with the following message in the log file:

This error originated from a custom integration.

Logger: custom_components.reolink_cctv.config_flow
Source: components/webhook/__init__.py:46
Integration: Reolink IP NVR/camera (documentation, issues)
First occurred: 9:40:25 AM (4 occurrences)
Last logged: 9:46:10 AM

Unexpected exception
Traceback (most recent call last):
  File "/config/custom_components/reolink_cctv/config_flow.py", line 76, in async_step_user
    self.info = await self.async_obtain_host_settings(self.hass, user_input)
  File "/config/custom_components/reolink_cctv/config_flow.py", line 142, in async_obtain_host_settings
    if not await host.init():
  File "/config/custom_components/reolink_cctv/host.py", line 187, in init
    if not await self.register_webhook():
  File "/config/custom_components/reolink_cctv/host.py", line 336, in register_webhook
    self._hass.components.webhook.async_register(DOMAIN, self._event_id, self._webhook_id, handle_webhook)
  File "/usr/src/homeassistant/homeassistant/components/webhook/__init__.py", line 46, in async_register
    raise ValueError("Handler is already defined!")
ValueError: Handler is already defined!

**Comment by JimStar** [comment]
Thank you for the info!  
I've removed **0.0.24**, use the **0.0.23** so far.

It looks like some regression introduced with new features. I"m just guessing things, not having all different models to test..:)  
Will find and fix this in a next version soon...

**Comment by JimStar** [comment]
Could you please post a full debug log when you get this error?

**Comment by apranda** [comment]
Unfortunately I turned off logging before I updated to 0.0.24 and reverted to 0.0.23. Is there a way I can reinstall 0.0.24 using HACS so I can collect the log later today?

**Comment by JimStar** [comment]
I've already deleted the **0.0.24** release, so only if installed manually from **master** trunk...

Can you remember, did you reboot HA after removing the integration entry and before trying to add a new one (where you got the error)?

**Comment by skynet01** [comment]
I also get this error when try to add the doorbell, both https and http. I still got 24 installed. How do you turn on debug

**Comment by JimStar** [comment]
You can switch on debug logging with this in the HA config:

```
logger:
  default: warning
  logs:
    custom_components.reolink_cctv: debug
    reolink_ip: debug
```

**Comment by JimStar** [comment]
Try to reboot HA before adding a new integration entries...

**Comment by apranda** [comment]
No, I did not reboot HA in between removing and adding back the camera when on 0.0.24

**Comment by skynet01** [comment]
Got it, here are the logs. I removed the Doorbell, restarted then added the debug config and rebooted again. Attached is the log file.  I have another reolink camera on there its on the 145 IP and the doorbell is on 148

https://codeshare.io/mpYlK4

**Comment by JimStar** [comment]
@skynet01  
Is this log already with the doorbell-camera added and rebooted, or is it when you're just trying to add a new integration entry?..

**Comment by skynet01** [comment]
Just when I tried to add it as a new integration entry right after I rebooted.

**Comment by JimStar** [comment]
@skynet01  
Just one side note, regarding what I see in your log... Not related to this bug, just for your information...  
Do you really have the 1-second watchdog interval set?.. You're overloading your HA resources and your network resources **a lot** with this, while with your cameras you don't need the watchdog at all. Your cameras send the rich ONVIF notifications, having all the data in them, so the integration receives all these notifications without problems... you don't need to send a lot of data back-and-forth to/from the cameras every one second...

**Comment by skynet01** [comment]
Oh please tell me more 😄 .. how do I change this watchdog setting to the correct interval and where do you see it? You are correct maybe it's something left over from the reolink_dev integration?

**Comment by JimStar** [comment]
You can safely set this interval to something like 60 or 120 seconds or so...  
Just look at the configuration window of each integration entry, there should be a watchdog interval setting. Look at what time it is set to. Do you see if this is set lower than 60 seconds somewhere?..

**Comment by skynet01** [comment]
Are you referring to the main watchdog component that checks if HA is running (or addons are working) or some specific watchdog  inside an integration? I know I never messed with any watchdog settings this is all defaults and warnings at the top are from home assistant starting up.

**Comment by JimStar** [comment]
No, it's not a HA watchdog. It's this integration specific watchdog, that checks from time to time if the subscription is still alive... if not - it then polls the devices, but **only** if the subscription failed... This setting is in this integration's config window.  
Actually, scratch this - I just see it is 1 minute set in your logs, not 1 second. :) This is a normal time... you can even have it more, like 3 minutes...

BTW, I've already found this bug's culprit, will release a fixed version soon... At least **I think** I've found it... :smile:

**Comment by skynet01** [comment]
Oh yes, I see it now. Updated it. Hope the debug helped with the fix

**Comment by JimStar** [comment]
Should be fixed in [0.0.25](https://github.com/JimStar/reolink_cctv/releases/tag/0.0.25).
